peerstore.go

```diff
@@ -75,13 +80,30 @@ func (ps *peerstore) PeerInfo(p peer.ID) PeerInfo {
 	}
 }
 
+func (ps *peerstore) internProtocol(s string) string {
+	if len(s) > maxInternedProtocolSize {
+		return s
+	}
+
+	if interned, ok := ps.internedProtocols[s]; ok {
+		return interned
+	}
+
+	if len(ps.internedProtocols) >= maxInternedProtocols {
+		ps.internedProtocols = make(map[string]string, maxInternedProtocols)
```

Comment:
not sure I understand this. If we have more than `maxInternedProtocols` we throw the map away? why?

Reply:
I guess it's "throw hands in the air" and start interning anew.

Reply:
Basically. This basically just drops the "cache" when it gets full.